ob1k-cache/src/main/java/com/outbrain/ob1k/cache/metrics/AsyncBulkGetMetrics.java

```diff
@@ -0,0 +1,42 @@
+package com.outbrain.ob1k.cache.metrics;
+
+import com.outbrain.ob1k.concurrent.Try;
+import com.outbrain.swinfra.metrics.api.Counter;
+import com.outbrain.swinfra.metrics.api.MetricFactory;
+import com.outbrain.swinfra.metrics.api.Timer;
+import java.util.Map;
+
+/**
+ * Holds a set of metrics used for measuring get operations.
+ *
+ * @author hunchback
+ */
+public class AsyncBulkGetMetrics extends AsyncOperationMetrics {
+  private final Counter hits;
+  private final Counter total;
+
+  public AsyncBulkGetMetrics(final MetricFactory metricFactory, final String component) {
+    super(metricFactory, component);
+    hits = metricFactory.createCounter(component, "cacheHits");
+    total = metricFactory.createCounter(component, "cacheTotal");
+  }
+
+  public void increaseTotal(final Iterable keys) {
+    for (final Object key : keys) {
+      total.inc();
```

Comment:
Performance wise, if you must spin here, better count locally, and update the counter once at the end, but I must say I don't like this loop here. You can use com.google.common.collect.Iterables.size() instead.